src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+    private double diagonal;
+
+    public Square(double side, double diagonal, String color) {
+        super(color);
+        this.side = side;
+        this.setDiagonal(diagonal);
+    }
+
+    @Override
+    public double area() {
+        double s;
+        s = (getDiagonal() * getDiagonal()) / 2;
```

Comment:
The formula for calculating the area of a square is incorrect. Use `side * side` instead of `(diagonal * diagonal) / 2`.